substrate/frame/revive/src/lib.rs

```diff
@@ -1781,6 +1783,74 @@ impl<T: Config> Pallet<T> {
 			.map(|code| code.into())
 			.unwrap_or_default()
 	}
+
+	/// Transfer a deposit from some account to another.
+	///
+	/// `from` is usually the transaction origin and `to` a contract or
+	/// the pallets own account.
+	fn charge_deposit(
+		hold_reason: Option<HoldReason>,
+		from: &T::AccountId,
+		to: &T::AccountId,
+		amount: BalanceOf<T>,
+		exec_config: &ExecConfig,
+	) -> DispatchResult {
+		use frame_support::traits::tokens::{Fortitude, Precision, Preservation};
+		match (exec_config.collect_deposit_from_hold, hold_reason) {
```

Comment:
If `exec_config.collect_deposit_from_hold` is `true`, then `from` is ignored. Is this intended – it is different to the original logic.

When a code is uploaded [in `store_code`](https://github.com/paritytech/polkadot-sdk/pull/9803/files#diff-feb8f2e6149e5fd5c5c6225036dd464fae9cb9dd016beb61d80fbc4ef36cf1c5R197), then `from` is not necessarily the transaction origin: if this is a contract creation call for EVM, `from` is the direct caller, which does not need to be the extrinsic origin.